src/Configuration/AccountManager.cs

```diff
@@ -9,42 +10,64 @@ internal static class AccountManager
     {
         public static List<Account> Accounts;
 
-        private static void Load(string serverName)
+        public static string[] GetAccountNames(string serverName)
         {
-            if (Accounts == null)
+            try
             {
-                var accounts = LoadAccountsFromFile();
-                Accounts = accounts.Where(x => x.Server == serverName).ToList();
+                Load(serverName);
+                return Accounts?.Where(x => x.Server == serverName).Select(x => x.UserName).ToArray() ?? new string[] { };
+            }
+            catch(Exception ex)
+            {
+                //Log error?
+                return new string[] { };
             }
-        }
-
-        public static string[] GetAccountNames(string serverName)
-        {
-            Load(serverName);
-            return Accounts?.Where(x => x.Server == serverName).Select(x => x.UserName).ToArray() ?? new string[] { };
         }
 
         public static string GetAccountPassword(string serverName, string userName)
         {
-            Load(serverName);
-            return Accounts?.FirstOrDefault(x => x.UserName == userName)?.Password;
+            try
+            {
+                Load(serverName);
+                return Accounts?.FirstOrDefault(x => x.UserName == userName)?.Password;
+            }
+            catch(Exception ex)
+            {
+                //Log Error?
+                return string.Empty;
+            }
         }
 
         public static void SaveAccount(string serverName, string userName, string password)
         {
-            Load(serverName);
-            var existingRecord = Accounts.FirstOrDefault(x => x.Server == serverName && x.UserName == userName);
-            if(existingRecord == null)
+            try
             {
-                Accounts.Add(new Account() { UserName = userName, Server = serverName, Password = password }); //TODO: Figure if the pw needs to be encrypted here or before.
+                Load(serverName);
+                var existingRecord = Accounts.FirstOrDefault(x => x.Server == serverName && x.UserName == userName);
+                if (existingRecord == null)
+                {
+                    Accounts.Add(new Account() { UserName = userName, Server = serverName, Password = password });
```

Comment:
Can you make `Account` immutable? ie `new Account(serverName, userName, password)`